src/Microsoft.Extensions.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -12,6 +12,17 @@ namespace Microsoft.Extensions.Configuration
     public interface IConfiguration
     {
         /// <summary>
+        /// Gets the key this element occupies in its parent or "" for the IConfigurationRoot.
+        /// </summary>
+        string Key { get; }
```

Comment:
@haok moving `Key` and `Path` to `IConfiguration` is a redesign of the interfaces and I don't think it belongs in this PR. If you want to start a discussion about redesigning the interfaces let's do that separately. 

For some context, `Key`, `Path` and `Value` are defined only for "child" `IConfiguration` (i.e. `IConfigurationSection`) because they felt very unnatural on the configuration root. This was raised by @davidfowl and others. I have raised the fact that it is not great to have to always try cast/is check `IConfiguration` instances in library code to see what they support but so far this is still the design.

FWIW, if end up changing the design, we should probably move `Value` as well and get rid of the `IConfigurationSection` interface altogether.